torch/onnx/_internal/exporter/_torchlib/ops/hop.py

```diff
@@ -92,3 +92,53 @@ def higher_order_cond(
             (), else_node.outputs, nodes=[else_node], name=false_func.name
         ),
     )
+
+
+@onnx_impl(torch.ops.higher_order.scan, no_compile=True)
+def higher_order_scan(
+    body_func: ir.Function,
+    scan_inits: Sequence[ir.Value],
+    scan_inputs: Sequence[ir.Value],
+    additional_inputs: Sequence[ir.Value] | None,
+    reverse: bool = False,
+) -> Sequence[ir.Value]:
+    node_inputs = [
+        *[
+            # inputs are clone to make sure they have different names
+            # from additional inputs which could be the same:
+            # one input can be a scanned input and an additional input
```

Comment:
Will this be removed by identity removal optimization? Curious why they need to be different? Is it a requirement of Scan?

Reply:
Identity removal may an issue with it. In some scenario, you need x as a scanned variable (x is a row then) and x as an input (the whole tensor). With torch, the get different name in the function implementing the body. onnx includes the local context so x could be rewritten by the scanned x if don't pay attention.

Reply:
Oh that's right. I missed that. The function's inputs should not be used in anywhere else. And all nodes need to belong to _some_ graph.